### PR TITLE
FOUR-10195: Collaborative UI components

### PR DESCRIPTION
**Pull request by rodriquelca**
ci:deploy
ci:modeler:next
ci:package-collections:develop
ci:package-data-sources:develop
ci:package-decision-engine:develop

## Issue & Reproduction Steps

Expected behavior: 
as a developer I want to have multiplayer avatar and client cursor component 
Actual behavior: 
n/a
## Solution
two new components was added:
- add multiplayer avatar
- add client cursor
avatar-curso

## How to Test
Test the steps above
- go to standalone modeler 
- npm run serve
- add the next code to the Modeler.vue file into the mounted section
`  if (process.env.NODE_ENV !== 'production') {
      this.players = [
        { id: '1',
          name: 'Juan Perez',
          color: '#FF6F61',
          imgSrc: 'https://placekitten.com/300/300',
          top: 90,
          left:80,
        },
        { id: '2',
          name: 'Ricardo Ford',
          color: '#34568B',
          top: 500,
          left:100,
        },
        { id: '3',
          name: 'Mauri Clear',
          color: '#88B04B', 
          top: 200,
          left:200,
        },
        { id: '4',
          name: 'Tisha Mccullough',
          color: '#92A8D1',
          imgSrc: 'https://xsgames.co/randomusers/avatar.php?g=female',
          top: 800,
          left: 600,
        },
      ];
    }`
## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-10195

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

### CODE REVIEW

File: src/components/modeler/Modeler.vue [hunk 1]
```diff
@@ -153,6 +154,15 @@
         @save-state="pushToUndoStack"
       />
     </b-row>
+
+    <RemoteCursor 
+      v-for="(player, index) in players"
+      :cursor-color="player.color"
+      :username="player.name"
+      :key="index"
```
Comment (josechirivella) [inline src/components/modeler/Modeler.vue:162]: do players have an id? Use that instead

File: src/components/topRail/multiplayerViewUsers/MultiplayerViewUsers.vue [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+<script>
+import Avatar from '@/components/topRail/multiplayerViewUsers/avatar/Avatar';
+export default {
+  components:{
+    Avatar,
+  },
+  props: {
+    players: {
+      type: Array,
+      required: [],
```
Comment (josechirivella) [inline src/components/topRail/multiplayerViewUsers/MultiplayerViewUsers.vue:18]: you mean `required: true`

File: src/components/topRail/multiplayerViewUsers/MultiplayerViewUsers.vue [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<template>
+  <b-avatar-group class="container">
+    <template v-for="(item, index) in players" >
+      <Avatar :badgeBackgroundColor="item.color" :imgSrc= "item.imgSrc" :userName="item.name" v-bind:key="index"/>
```
Comment (josechirivella) [inline src/components/topRail/multiplayerViewUsers/MultiplayerViewUsers.vue:4]: same here. the `:key` should be the player id if available

File: src/components/topRail/multiplayerViewUsers/avatar/Avatar.vue [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    },
+  },
+  methods: {
+    /**
+     * Get the initials from a given name.
+     *
+     * @param {string} name - The full name from which to extract initials.
+     * @returns {string} The initials of the first and last names.
+     */
+    getInitials(name= '') {
```
Comment (josechirivella) [inline src/components/topRail/multiplayerViewUsers/avatar/Avatar.vue:60]:
```suggestion
    getInitials(name = '') {
```

**Review by josechirivella — CHANGES_REQUESTED** [review]

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-5e6aee50e9.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-5e6aee50e9.eng.processmaker.net

**Review by josechirivella — APPROVED** [review]

**Review by devmiguelangel — APPROVED** [review]

**Pull request merged**